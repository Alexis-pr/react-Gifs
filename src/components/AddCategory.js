import React, { useState } from 'react'
import PropTypes from 'prop-types';

const AddCategory = ( {setCategories}) => {
    const [inputValue, setInputValue] = useState('')

    const handleChange=(e)=>{
        setInputValue(e.target.value)
        console.log('HandleChange fue llamado desde addCategory')
    }
    const handleSubmit =(e)=>{
        e.preventDefault()
        console.log('esto es un llamado desde el handleSubmit de addCategory')
        if( inputValue.trim().length >= 3){
          setCategories(categor => [inputValue , ...categor ] )
          setInputValue('')
        } else{
            alert('Tiene q ser mayor a 3 letras')
        } 
            
    }

    return (
        <>
         {/*    <h2>Buscar...</h2>   */}
            <form onSubmit={handleSubmit}>
                <p>{inputValue}</p>
                <input 
                    type='text'
                    value={inputValue}
                    onChange={handleChange}
                    placeholder="Buscar...."
                />
            </form>
             
        </>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
export default AddCategory
