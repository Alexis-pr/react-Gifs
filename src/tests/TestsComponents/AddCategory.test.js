import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en el componente AddCategory', () => {
    const setCategories = jest.fn()
    let wrapper = shallow(  < AddCategory setCategories={setCategories} />)
    beforeEach( () => {
        jest.clearAllMocks()
         wrapper = shallow(  < AddCategory setCategories={setCategories} />)
    })

    test('Debe mostrarse Correctamente ', () => {
        expect(wrapper).toMatchSnapshot()        
    })

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input')
        /* console.log(input)  */  
        const value = 'Hola mundo'
        input.simulate('change',{ target : { value:value } } )

        expect( wrapper.find('p').text().trim() ).toBe(value)
    })

    test('No debe de imprimir o mostrar la informacion Onsubmit', () => {
        wrapper.find('form').simulate('submit',{ preventDefault(){} })
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
            const value = 'Hola mundo'
            wrapper.find('input').simulate('change',{ target : { value:value } } )
            
            wrapper.find('form').simulate('submit',{ preventDefault(){} })
            expect(setCategories).toHaveBeenCalled(); // verifica que halla pasado al menos una vez
            expect(setCategories).toHaveBeenCalledTimes(1) //verifica las veces q se ejecuta
            expect(setCategories).toHaveBeenCalledWith(expect.any(Function)) //verifica que el parametro ingresado sea una funcion
           /*  wrapper.find('input').simulate('change',{ target : { value: ' ' } } ) */  
           /* Esta es la forma que la hice yo, Ã±p que hice fue llamar de nuevo el wrapper y darle un valor nuevo al value:value, de esta forma
           el segundo value igualandolo a un string vacio */
           expect( wrapper.find('input').prop('value') ).toBe('')

    })
    

    
})
