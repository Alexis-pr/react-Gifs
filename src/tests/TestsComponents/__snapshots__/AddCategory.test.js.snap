// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe mostrarse Correctamente  1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      placeholder="Buscar...."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
